### ISSUE TITLE
Glibc-2.24 available upstream

### ISSUE THREAD

**Issue by lancelotsix** [NixOS/nixpkgs#17662]
## Issue description

Glibc-2.24 is available upstream. It contains security fixes (most have been backported into nixpkgs:glibc-2.23).

I began the process of updating the derivation but face problems with updated locales build handeling in https://github.com/lancelotsix/nixpkgs/commit/599cdf79d8b5047b9550ddf252d7e3c83b734bf3.

I’ll be glad if anyone sees something obvious I missed to build the `glibc` and `glibLocales` derivations before I try to dive into glibc build system.

cc maintainer @edolstra 

### Steps to reproduce

```
$ nix-build -A glibc
…
.././scripts/mkinstalldirs /var/run/current-system/sw/lib/locale
mkdir -p -- /var/run/current-system/sw/lib/locale
mkdir: cannot create directory '/var': Permission denied
Makefile:157: recipe for target '/var/run/current-system/sw/lib/locale/locale.alias' failed
make[2]: *** [/var/run/current-system/sw/lib/locale/locale.alias] Error 1
make[2]: Leaving directory '/tmp/nix-build-glibc-2.24.drv-0/glibc-2.24/intl'
Makefile:214: recipe for target 'intl/subdir_install' failed
make[1]: *** [intl/subdir_install] Error 2
make[1]: Leaving directory '/tmp/nix-build-glibc-2.24.drv-0/glibc-2.24'
Makefile:12: recipe for target 'install' failed
make: *** [install] Error 2
builder for ‘/nix/store/239yngndbadn7wx9d0vxcp9lmnmssrzj-glibc-2.24.drv’ failed with exit code 2
error: build of ‘/nix/store/239yngndbadn7wx9d0vxcp9lmnmssrzj-glibc-2.24.drv’ failed
```

## Technical details

* System: 16.03.1111.f231b2b (Emu)
* Nix version: nix-env (Nix) 1.11.2
* Nixpkgs version: https://github.com/lancelotsix/nixpkgs/tree/update_glibc

**Comment by RamKromberg** [comment]
I wasn't about to say anything, but seeing how you're already did gcc 4.9.4 (Thanks!) and are now doing glibc, I'd be more worried about gcc 5.3. Unlike branch 4 and 4.9 specifically which still receives backported patches, GCC has stopped issuing tertiary releases in branch 5 and will not backport fixes withing branch 5 (https://gcc.gnu.org/develop.html).

That is, there won't be 5.3.1.

Already, multiple security issues were identified and corrected in 5.4 ( CVE-2016-2226 CVE-2016-4488 CVE-2016-4489 CVE-2016-4490 ...) and are not mentioned in the release logs or are searchable in bugzilla in fear of being exploited by blackhats ( compare https://gcc.gnu.org/bugzilla/buglist.cgi?bug_status=RESOLVED&query_format=advanced&resolution=FIXED&target_milestone=5.4&order=changeddate%20DESC%2Cbug_status%2Cpriority%2Cassigned_to%2Cbug_id&query_based_on= versus https://gcc.gnu.org/bugzilla/show_bug.cgi?id=69687 ).

Anyhow, I won't open a separate issue over this since the people who need to know about it, probably already do. But since you're already putting in the effort...

And again, thanks for 4.9.4 :P

**Comment by FRidh** [comment]
`forkpty` patch is not needed anymore with 2.24 (https://github.com/NixOS/nixpkgs/pull/12895#issuecomment-239779697)

**Comment by vcunat** [comment]
@RamKromberg: our gcc is 5.4.0 on master, but we still have 5.3.0 on 16.03. That could be backported, though I remember we did have to fixup a few packages due to some math include changes.

**Comment by vcunat** [comment]
@lancelotsix: I don't think sandboxed builders can even see `/var/run/current-system`. I'd try with sandboxing first, although it will certainly be a plus if it builds even without it.

**Comment by RamKromberg** [comment]
@vcunat yeah sorry @globin already corrected me on this in the hardening discussion. No point backporting to 16.03 with 16.09 on the way though... no?

**Comment by vcunat** [comment]
@RamKromberg: if they're not critical, they're not worth the work, at least for myself ATM.

**Comment by RamKromberg** [comment]
@vcunat That fine regarding individual patches. But I'm still concerned about being a point release behind current considering all the un-loged issues... Still, so long as 16.09 will have 5.4, no problem :D

**Comment by vcunat** [comment]
It certainly will; 16.09 hasn't even been branched from master yet.

**Comment by vcunat** [comment]
Resolved in staging, apparently: 9744c7768,
